### DISCUSSION TITLE
AWS setup

### DISCUSSION THREAD

**Discussion by MananpreetKaur1** [h2oai/h2o-llmstudio#649] [Q&A]
I have been trying to setup llm studio and all the installations before the make setup command works fine but as I try with 'make setup' it always shows me the following error 'make: *** No rule to make target 'setup'.  Stop.'.
I have tried installing make and it shows that its already installed. Can someone please help me with this?
And maybe provide with more details on the setup of LLM studio on ec2 instance G4 series??????

**Comment by psinger** [comment]
The setup on an AWS vm should be exactly as on a local machine and described in the `README`. Are you sure you are in the main folder when running the `make setup` command?

**Reply by MananpreetKaur1** [reply]
What is the main folder? I am in local user directory

**Reply by MananpreetKaur1** [reply]
@psinger - I am logged in as llmstudio and in home folder /home/ubuntu ? Can you please guide on which is the main folder

**Reply by MananpreetKaur1** [reply]
I was able to solve this but now during execution make setup getting below error:
RuntimeError: failed to build image pip because:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/virtualenv/seed/embed/via_app_data/via_app_data.py", line 57, in _install
    installer.install(creator.interpreter.version_info)
  File "/opt/conda/lib/python3.10/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 33, in install
    self._uninstall_previous_version()
  File "/opt/conda/lib/python3.10/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 144, in _uninstall_previous_version
    dist_name = self._dist_info.stem.split("-")[0]
  File "/opt/conda/lib/python3.10/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 110, in _dist_info
    raise RuntimeError(msg)  # pragma: no cover
RuntimeError: no .dist-info at /home/ubuntu/.local/share/virtualenv/wheel/3.10/image/1/CopyPipInstall/pip-24.0-py3-none-any, has pip

**Reply by psinger** [reply]
Can you please follow the README as provided? It seems you are leaving some steps out, like installing proper python packaging:
https://github.com/h2oai/h2o-llmstudio?tab=readme-ov-file#system-installs-python-310

I would also suggest to consult Google or LLMs for these types of system specific errors.

If a local install is too complicated, we provide an out-of-the-box Docker based solution you can use:
https://github.com/h2oai/h2o-llmstudio?tab=readme-ov-file#run-h2o-llm-studio-gui-using-docker-from-a-nightly-build

**Reply by MananpreetKaur1** [reply]
Thanks for all the links. There was an error with the virtual environment build which I learnt about from https://jdsalaro.com/howto/fix-pipenv-runtimeerror-no-dist-info-has-pip/ blog. Following your suggestion, I proceeded to reinstall Python, which resolved the issue.
However, I encountered another obstacle when utilizing AWS EC2 instance was for less space allocated to root directory (that is, by default it comes with just 45GiB of storage in the root directory for Ubuntu (20.04) AMI with preinstalled PyTorch(2.1.0) and NVIDIA Drivers, whereas the whole installation process requires at least 60 GiB of storage in the root directory (58 was exactly used in my machine after the installation). 
Also, please ensure that you add the storage value to the root volume and not use the 'add new volume' option given in the configure storage tab in the launch instance page for the same.
If you end up still starting up your instance with the default storage setting AWS provides the option to increase the volume. However, post-resizing, it's crucial to locate where the added space is reflected in the mounted files on the machine and subsequently resize the root directory using the 'sudo growpart / 1' command. A restart of the machine is required, but it's essential to perform a Stop and Start instead of a simple reboot to ensure that the added space correctly reflects in the root directory.
One other error I faced in the last was in launching the application after following all the steps successfully as given in the README file here, which was because I was referring my aws machine on my personal machine browser so one on your working AWS machine give your machine access to the required ports for the application in the Security Tab by editing the Inbound Rules by entering the default security groups in the same window. If you don't know the port or type of protocols then in the add rule you can simply choose the Type = All Traffic, Source = Custom and 0.0.0.0/0 for setting the access on at all the ports for the time being. Secondly, take your machine's Public IPv4 DNS as shown in the instances page and replace it with localhost in the URL and then try accessing it.
